### PR TITLE
Feature/registration data binding

### PR DESCRIPTION
**Pull request by HARSHKALAL**

### CODE REVIEW

File: Chatify/chat/api.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 from .serializers import UserSerializer
 from .models import User
+from .utils import *
 
 
 class RegistrationApi(APIView):
     def post(self, request):
         try:
-            serializer = UserSerializer(data=request.data)
+            serializer = UserSerializer(data=set_contact_number(request.data))
```
Comment (yatishTrootech) [inline Chatify/chat/api.py]: phone number should validated from  ```validate_phone_nuber``` method

File: Chatify/chat/static/js/ajax_request.js [hunk 2]
```diff
@@ -8,13 +7,13 @@ function makeAjaxRequest(methodType, csrfToken, url, data, callback)
...
-        success: function(data) {
+        success: function (data) {
             if (callback) {
                 callback(data)
             }
         },
-        error: function(data) {
-           console.log(data)
+        error: function (data) {
+            console.log(data)
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/ajax_request.js:16]: Show error dont console

File: Chatify/chat/static/js/register.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 function register(e) {
         e.preventDefault();
+        console.log("Ajax call")
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/register.js]: remove console @Dhruvi-Makwana Why you have not added comment ?

File: Chatify/chat/static/js/register.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
         var csrfToken = $('input[name="csrfmiddlewaretoken"]').val();
         var formdata = new FormData($('#register_form')[0]);
-        makeAjaxRequest('POST', csrfToken, "/register/", formdata, function(response) {
+        makeAjaxRequest('POST', csrfToken, "http://127.0.0.1:8000/api/register/", formdata, function (response) {
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/register.js]: Why full URL? @Dhruvi-Makwana Why comment not added by you ?

File: Chatify/chat/utils.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from rest_framework import serializers
+from rest_framework import status
+
+
+def set_contact_number(data):
```
Comment (yatishTrootech) [inline Chatify/chat/utils.py]: def validate_phone_number

File: Chatify/chat/utils.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    get_mobile_number = data.get("mobile_number")
```
Comment (yatishTrootech) [inline Chatify/chat/utils.py]: wrong variable name

File: Chatify/chat/utils.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    if len(get_mobile_number) > 10 and len(get_mobile_number) <= 13:
+        if get_mobile_number[:2] == "91":
+            data["mobile_number"] = "+" + get_mobile_number
+        elif get_mobile_number[:3] != "+91":
+            raise serializers.ValidationError(
+                {"phone_number": "Phone number is Not Valid"}
+            )
+    elif len(get_mobile_number) == 10:
```
Comment (yatishTrootech) [inline Chatify/chat/utils.py]: it will fail for "+912332323"

File: Chatify/chat/utils.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        data["mobile_number"] = "+91" + get_mobile_number
```
Comment (yatishTrootech) [inline Chatify/chat/utils.py]: what if i pass 20 length mobile number ?

**Review by yatishTrootech — CHANGES_REQUESTED** [review]

File: Chatify/chat/serializers.py [hunk 9]
```diff
@@ -36,3 +37,7 @@ def validate_password(self, password):
             if password != confirmation_password:
                 raise serializers.ValidationError("PASSWORD DOESNOT MATCH")
         return password
+
+    def validate_mobile_number(self, mobile_number):
+        breakpoint()
```
Comment (yatishTT) [inline Chatify/chat/serializers.py]: why breakpoint is there ?>

File: Chatify/chat/serializers.py [hunk 10]
```diff
@@ -36,3 +37,7 @@ def validate_password(self, password):
...
+        return mobile_number
```
Comment (yatishTT) [inline Chatify/chat/serializers.py]: return validate_contact_number(mobile_number) from utils

File: Chatify/chat/serializers.py [hunk 11]
```diff
@@ -37,3 +37,7 @@ def validate_password(self, password):
             if password != confirmation_password:
                 raise serializers.ValidationError("PASSWORD DOESNOT MATCH")
         return password
+
+    def validate_mobile_number(self, mobile_number):
+        breakpoint()
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/serializers.py]: remove breakpoint()

File: Chatify/chat/urls.py [hunk 12]
```diff
@@ -4,7 +4,7 @@
 
 
 urlpatterns_api = [
-    path("api/register/", RegistrationApi.as_view(), name="RegistrationApi"),
+    path("api/register/", RegistrationApi.as_view(), name="api/register/"),
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/urls.py]: is it correct name for this url ? don;t we simply write "RegistrationApi".

**Review by yatishTT — APPROVED** [review]

**Pull request merged**